### PR TITLE
Assignment #7

### PR DESCRIPTION
**Pull request by wortleyjk-codefellows**
added .map() and localStorage caching checks in display-data.js

### CODE REVIEW

**Comment by meganwalter** [comment]
Merge these please

File: javascript/display-data.js [hunk 1]
```diff
@@ -28,16 +27,28 @@ var populateBlogFilter = function(elem){
...
 
-Data.populatePage = function(){
-  $.get('data/data.json',function(data){
-    data.forEach(function(elem){
-      blogCollection.push(elem);
+Data.populateBlogPage = function(){
+  $.get('data/data.json',function(data,text,xhr){
+    var value = [];
+    responseETag = xhr.getResponseHeader('Etag');
+    if(localStorage.getItem('eTag')===null||responseETag!=localStorage.getItem('eTag')){
```
Comment (meganwalter) [inline javascript/display-data.js:22]: indenting is important to see where a function begins and ends (just for fyi)

File: javascript/events.js [hunk 2]
```diff
@@ -43,23 +43,27 @@ var enableTeaserLinks = function(){
...
-//show About
-$('a.ion-ios-person').on('click',function(e){
+  }
+  else if($(this).hasClass('ion-home')){
     e.preventDefault();
     $('section').hide();
-    $('#about-me').slideDown(500)
+    $('#home-wrapper').slideDown(400);
+  }
 });
```
Comment (meganwalter) [inline javascript/events.js:65]: There is a way to do this last function with less code! I can show you if you want!

File: javascript/events.js [hunk 3]
```diff
@@ -43,23 +43,27 @@ var enableTeaserLinks = function(){
...
-});
-//show About
-$('a.ion-ios-person').on('click',function(e){
+  }
+  else if($(this).hasClass('ion-home')){
     e.preventDefault();
     $('section').hide();
-    $('#about-me').slideDown(500)
+    $('#home-wrapper').slideDown(400);
+  }
```
Comment (meganwalter) [inline javascript/events.js:63]: Where is your IIFE??

File: javascript/display-data.js [hunk 4]
```diff
@@ -28,16 +27,28 @@ var populateBlogFilter = function(elem){
...
+      localStorage.setItem('eTag',responseETag);
+      localStorage.setItem('blogArticles',JSON.stringify(data));
+      value = data;
+    }
+    else{
+      value = JSON.parse(localStorage.getItem('blogArticles'));
+    }
+    var blogCollection = [];
+    value.map(function(elem){
```
Comment (meganwalter) [inline javascript/display-data.js]: This is basically a forEach, re-written as a .map function

disregard previous,
Dan says it's legit, and a good use of map.

Comment (wortleyjk-codefellows) [inline javascript/display-data.js]: understood, will modify this right away

Comment (meganwalter) [inline javascript/display-data.js]: Don't worry, dan says its legit

File: javascript/events.js [hunk 5]
```diff
@@ -43,23 +43,24 @@ var enableTeaserLinks = function(){
...
-});
-//show About
-$('a.ion-ios-person').on('click',function(e){
+  }
+  else if($(this).hasClass('ion-home')){
     e.preventDefault();
-    $('section').hide();
-    $('#about-me').slideDown(500)
+    $('#home-wrapper').slideDown(400);
+  }
```
Comment (meganwalter) [inline javascript/events.js:63]: Where is the IIFE?

Comment (wortleyjk-codefellows) [inline javascript/events.js:63]: IIFE starts on Line 2 in display-data.js.

**Pull request merged**